### PR TITLE
add gtid set information in query event

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
add gtid set information in query event

### CODE REVIEW

File: replication/event.go [hunk 1]
```diff
@@ -241,6 +241,9 @@ type QueryEvent struct {
 	StatusVars    []byte
 	Schema        []byte
 	Query         []byte
+
+	// in fact QueryEvent dosen't have the GTIDSet information, just for beneficial to use
```
Comment (siddontang) [inline replication/event.go]: what is the benefit?

Comment (GregoryIan) [inline replication/event.go]: As `Gset` in `XID Event`,   `Query Event` ( DDL)  don't have `XID Event`,  so we need `Gset` to manage save point  by ourself

**Comment by GregoryIan** [comment]
LGTM

File: replication/event.go (hunk 1 above)
Comment (siddontang) [inline replication/event.go]: when do you set GTIDSet to the Query event?

**Pull request merged**